core/src/main/java/org/fao/geonet/kernel/datamanager/base/BaseMetadataStatus.java

```diff
@@ -0,0 +1,151 @@
+package org.fao.geonet.kernel.datamanager.base;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang.StringUtils;
+import org.fao.geonet.constants.Params;
+import org.fao.geonet.domain.Group;
+import org.fao.geonet.domain.ISODate;
+import org.fao.geonet.domain.MetadataStatus;
+import org.fao.geonet.domain.MetadataStatusId;
+import org.fao.geonet.domain.MetadataStatusId_;
+import org.fao.geonet.domain.MetadataStatus_;
+import org.fao.geonet.kernel.datamanager.IMetadataIndexer;
+import org.fao.geonet.kernel.datamanager.IMetadataStatus;
+import org.fao.geonet.kernel.setting.SettingManager;
+import org.fao.geonet.kernel.setting.Settings;
+import org.fao.geonet.repository.GroupRepository;
+import org.fao.geonet.repository.MetadataStatusRepository;
+import org.fao.geonet.repository.SortUtils;
+import org.fao.geonet.repository.StatusValueRepository;
+import org.fao.geonet.repository.specification.MetadataStatusSpecs;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.data.domain.Sort;
+
+import jeeves.server.context.ServiceContext;
+
+public class BaseMetadataStatus implements IMetadataStatus {
+
+    @Autowired
+    private MetadataStatusRepository metadataStatusRepository;
+    @Autowired
+    private IMetadataIndexer metadataIndexer;
+    @Autowired
+    private StatusValueRepository statusValueRepository;
+    @Autowired
+    private GroupRepository groupRepository;
+    @Autowired
+    @Lazy
+    private SettingManager settingManager;
+
+    public void init(ServiceContext context, Boolean force) throws Exception {
+//        metadataStatusRepository = context.getBean(MetadataStatusRepository.class);
```

Comment:
I think `@Autowired` will not work due to multinode. Please check to use again `context.getBean`

Reply:
Fixed